azure-pipelines.yml

```diff
@@ -3,19 +3,112 @@
 #
 ---
 trigger:
-- master
+  branches:
+    include:
+      - master
+  tags:
+    include:
+      - "*"
 
-pool:
-  vmImage: "ubuntu-latest"
-
-steps:
-- task: NodeTool@0
-  inputs:
-    versionSpec: "12.x"
-  displayName: Use Node.js 12.x
-- script: npm ci
-  displayName: Install dependencies
-- script: npm audit
-  displayName: Audit dependencies
-- script: npm test
-  displayName: Run unit tests
+stages:
+  - stage: unit_tests
+    displayName: Unit tests
+    jobs:
+      - job: unit_tests
+        displayName: Unit tests
+        strategy:
+          matrix:
+            Linux:
+              imageName: "ubuntu-latest"
+            macOS:
+              imageName: "macOS-latest"
+            Windows:
+              imageName: "windows-latest"
+        pool:
+          vmImage: $(imageName)
+        steps:
+          - task: NodeTool@0
```

Comment:
you should put some of these steps in a template so you don't have to keep repeating them

Reply:
Done!